### ISSUE TITLE
Update Optimize module datapoints for consistency

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#451]
Datapoints in the Optimize module should receive minor updates as outlined in the ACs.

Implementation brief should include which pieces of JS need to be updated and which e2e tests need to be updated (if any).

**Execution is blocked until #406 is merged.**

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Optimize setup and querying Optimize data via API should still work as before.
* Parameters and return values controlled by the plugin itself should all be in camel case.
* Datapoints should be updated as follows:
    * GET optimize-id → keep as is
    * GET amp-client-id-opt-in → keep as is
    * GET amp-experiment-json → keep as is
    * POST optimize-id → keep as is
    * POST amp-experiment-json → keep as is
    * POST save → POST settings

#### [`POST save`]
* Rename datapoint to `settings`
    - `Optimize::get_datapoint_services`
    - `Optimize::create_data_request`
    - `OptimizeTest`
* Update usage
    * https://github.com/google/site-kit-wp/blob/b225df85d49c114e161591c91bbdb6a3c8015995/assets/js/modules/optimize/setup.js#L143
* No changes necessary for consistency with camel case

## Changelog entry

* Improve datapoint behavior of Optimize module for consistency.

**Comment by felixarntz** [comment]
IB ✅

**Comment by adamsilverstein** [comment]
This looks good in my testing. 

To test I set up analytics then optimize. Viewing the API requests from optimize when I change settings on the settings screen, I see the correct 'settings' url:

![image](https://user-images.githubusercontent.com/2676022/64539894-8820cd80-d2dc-11e9-8f94-7cf3b33c20f1.png)

In addition, I verified that updating/changing the optimize ID on the settings screen works as expected.

![image](https://user-images.githubusercontent.com/2676022/64540041-c74f1e80-d2dc-11e9-9207-78b0488c14ba.png)


Finally, I verified that the correct tagging appeared on the site front end:

![image](https://user-images.githubusercontent.com/2676022/64540130-ef3e8200-d2dc-11e9-990d-2e78f23fc08b.png)

Everything checks out here, moving to the approval column.
